### DISCUSSION TITLE
Future Plans?

### DISCUSSION THREAD

**Discussion by notwhoyouexpected** [Team-Tea-Time/laravel-forum#369] [General]
Hi, what are your future plans with this package? Any Features planned?

I Think it would be a great future if we could extend the package somehow so we could create "Addons" for it

best regards

**Comment by Riari** [comment]
Hi! I do have some features planned, yes. There isn't really a concrete roadmap or anything along those lines, but the ['type: feature' issues](https://github.com/Team-Tea-Time/laravel-forum/issues?q=is%3Aopen+is%3Aissue+label%3A%22type%3A+feature%22) will give you an idea of what I have in mind next.

Regarding your suggestion for addons, it's already possible to do quite a lot just by following Laravel conventions. You can implement event handlers, add your own routes, use the package's REST API, etc. The new [UI Presets](https://www.teamteatime.net/docs/laravel-forum/6.x/front-end/ui-presets/) feature is also a good way to build custom UIs.

I do need to think about ways to make it easier to augment existing functionality (for example, adding custom fields to the existing Create Thread form). Historically, I think people have achieved this by overriding the controller namespace for the frontend routes (see relevant discussion: https://github.com/Team-Tea-Time/laravel-forum/discussions/370) - that's no longer possible in version 6, but I'll either bring that back or implement a way of extending the built-in forms/actions.
